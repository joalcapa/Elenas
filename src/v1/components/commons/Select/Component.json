{
  "description": "",
  "displayName": "Component",
  "methods": [],
  "props": {
    "value": {
      "type": {
        "name": "string"
      },
      "required": false,
      "description": "Valor del select.",
      "defaultValue": {
        "value": "''",
        "computed": false
      }
    },
    "options": {
      "type": {
        "name": "array"
      },
      "required": false,
      "description": "Opciones del select.",
      "defaultValue": {
        "value": "[]",
        "computed": false
      }
    },
    "onSelect": {
      "type": {
        "name": "func"
      },
      "required": false,
      "description": "Funcion que devuelve el dato seleccionado.",
      "defaultValue": {
        "value": "() => {}",
        "computed": false
      }
    },
    "showSelect": {
      "type": {
        "name": "func"
      },
      "required": false,
      "description": "Funcion que muestra el selector (Modal).",
      "defaultValue": {
        "value": "() => {}",
        "computed": false
      }
    },
    "hiddenSelect": {
      "type": {
        "name": "func"
      },
      "required": false,
      "description": "Funcion que oculta el selector (Modal).",
      "defaultValue": {
        "value": "() => {}",
        "computed": false
      }
    },
    "isVisibleSelect": {
      "type": {
        "name": "bool"
      },
      "required": false,
      "description": "Determina si el selector se encuentra visible.",
      "defaultValue": {
        "value": "false",
        "computed": false
      }
    }
  }
}
