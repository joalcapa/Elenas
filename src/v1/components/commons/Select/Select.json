{
  "description": "Este componente renderiza un select basico.",
  "displayName": "Select",
  "methods": [],
  "props": {
    "value": {
      "type": {
        "name": "string"
      },
      "required": false,
      "description": "Valor del select.",
      "defaultValue": {
        "value": "''",
        "computed": false
      }
    },
    "options": {
      "type": {
        "name": "array"
      },
      "required": false,
      "description": "Opciones del select.",
      "defaultValue": {
        "value": "[]",
        "computed": false
      }
    },
    "onSelect": {
      "type": {
        "name": "func"
      },
      "required": false,
      "description": "Funcion que devuelve el dato seleccionado.",
      "defaultValue": {
        "value": "() => {}",
        "computed": false
      }
    }
  }
}
