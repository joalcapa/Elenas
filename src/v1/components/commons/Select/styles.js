import {StyleSheet, Dimensions} from 'react-native';

import {Colors} from '../../../resources';

const {width, height} = Dimensions.get('window');

const styles = StyleSheet.create({
    button_content: {
        width: 200,
        height: 40,
        borderWidth: 1,
        borderColor: Colors.C_7e807f,
        display: 'flex',
        justifyContent: 'center',
        paddingLeft: 20,
        borderRadius: 5,
    },
    content: {
        width: 300,
        height: 250,
        backgroundColor: Colors.C_FFFFFF,
    },
    item_content: {
        paddingTop: 20,
        paddingBottom: 20,
        paddingLeft: 20,
    },
    container: {
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: 'rgba(0,0,0,.5)',
        width,
        height,
    },
});

export default styles;