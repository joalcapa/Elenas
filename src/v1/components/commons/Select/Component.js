import React from 'react';
import PropTypes from 'prop-types';
import {TouchableOpacity, Modal, View, ScrollView} from 'react-native';

import TextField from '../../../components/commons/TextField';
import Button from '../../../components/commons/Button';

import styles from './styles';

/**
 * Componente que renderiza los items de un select basico.
 */
const Component = (props) => (
    <>
        <Modal
            animationType="none"
            transparent={true}
            visible={props.isVisibleSelect}
            onRequestClose={props.hiddenSelect}>
            <View style={styles.container}>
                <View style={styles.content}>
                    <ScrollView>
                        {props.options.map((option) => (
                            <TouchableOpacity
                                key={`Select${option.label}`}
                                style={styles.item_content}
                                onPress={() => {
                                    props.onSelect(option);
                                    props.hiddenSelect();
                                }}>
                                <TextField text={option.label} />
                            </TouchableOpacity>
                        ))}
                    </ScrollView>
                </View>
                <Button
                    isGreenButton
                    text={'Salir'}
                    onPress={props.hiddenSelect}
                />
            </View>
        </Modal>
        <TouchableOpacity onPress={props.options.length > 0 ? props.showSelect : null} style={styles.button_content}>
            <TextField text={props.value ? props.value : 'Seleccionar'} />
        </TouchableOpacity>
    </>
);

Component.propTypes = {
    /**
     * Valor del select.
     */
    value: PropTypes.string,
    /**
     * Opciones del select.
     */
    options: PropTypes.array,
    /**
     * Funcion que devuelve el dato seleccionado.
     */
    onSelect: PropTypes.func,
    /**
     * Funcion que muestra el selector (Modal).
     */
    showSelect: PropTypes.func,
    /**
     * Funcion que oculta el selector (Modal).
     */
    hiddenSelect: PropTypes.func,
    /**
     * Determina si el selector se encuentra visible.
     */
    isVisibleSelect: PropTypes.bool,
};

Component.defaultProps = {
    value: '',
    options: [],
    onSelect: () => {},
    showSelect: () => {},
    hiddenSelect: () => {},
    isVisibleSelect: false,
};

export default Component;