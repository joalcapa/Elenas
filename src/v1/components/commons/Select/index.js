import React from 'react';
import PropTypes from 'prop-types';
import {TouchableOpacity, Modal, View, ScrollView} from 'react-native';

import TextField from '../../../components/commons/TextField';
import Button from '../../../components/commons/Button';
import useSelect from '../../../hooks/useSelect';

import styles from './styles';

/**
 * Este componente renderiza un select basico.
 */
const Select = (props) => {
    const {onSelect, value, options} = props;
    const {isVisibleSelect, showSelect, hiddenSelect} = useSelect();

    return (
        <>
            <Modal
                animationType="none"
                transparent={true}
                visible={isVisibleSelect}
                onRequestClose={hiddenSelect}>
                <View style={styles.container}>
                    <View style={styles.content}>
                        <ScrollView>
                            {options.map((option) => (
                                <TouchableOpacity
                                    key={`Select${option.label}`}
                                    style={styles.item_content}
                                    onPress={() => {
                                        onSelect(option);
                                        hiddenSelect();
                                    }}>
                                    <TextField text={option.label} />
                                </TouchableOpacity>
                            ))}
                        </ScrollView>
                    </View>
                    <Button
                        isGreenButton
                        text={'Salir'}
                        onPress={hiddenSelect}
                    />
                </View>
            </Modal>
            <TouchableOpacity onPress={options.length > 0 ? showSelect : null} style={styles.button_content}>
                <TextField text={value ? value : 'Seleccionar'} />
            </TouchableOpacity>
        </>
    );
};

Select.propTypes = {
    /**
     * Valor del select.
     */
    value: PropTypes.string,
    /**
     * Opciones del select.
     */
    options: PropTypes.array,
    /**
     * Funcion que devuelve el dato seleccionado.
     */
    onSelect: PropTypes.func,
};

Select.defaultProps = {
    value: '',
    options: [],
    onSelect: () => {},
};

export default Select;