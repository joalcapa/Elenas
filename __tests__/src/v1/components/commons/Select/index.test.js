import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import {create} from 'react-test-renderer';
import {TouchableOpacity} from 'react-native';
import {shallow, configure} from 'enzyme';

import Select from '../../../../../../src/v1/components/commons/Select/Component';

configure({adapter: new Adapter()});
describe('Select component', () => {
    test('render Select component', () => {
        const tree = create(<Select />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('render Select component with value', () => {
        const tree = create(<Select value="select"/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('render Select component with options', () => {
        const tree = create(<Select options={[{label: 'option 1', value: '1'}, {label: 'option 2 ', value: '2'}]}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('onSelect functionality', () => {
        const onPressEvent = jest.fn();
        onPressEvent.mockReturnValue('Link on press invoked');
        const wrapper = shallow(<Select onSelect={onPressEvent} options={[{label: 'option 1', value: '1'}, {label: 'option 2 ', value: '2'}]} />);

        wrapper.find(TouchableOpacity).first().props().onPress();
        expect(onPressEvent.mock.results[0].value).toBe('Link on press invoked');
    });
});