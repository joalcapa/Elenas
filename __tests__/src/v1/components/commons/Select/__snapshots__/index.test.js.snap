// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select component render Select component 1`] = `
<div
  className="css-view-1dbjc4n r-borderColor-4yh6xw r-borderRadius-1f0042m r-borderWidth-rs99b7 r-cursor-1loqt21 r-display-6koalj r-justifyContent-1777fci r-maxWidth-127gp16 r-paddingBottom-1inuy60 r-paddingLeft-1yt7n81 r-paddingRight-ry3cjt r-paddingTop-glunga r-touchAction-1otgn73 r-transitionProperty-1i6wzkk r-userSelect-lrvibr r-width-13qz1uu"
  onClick={[Function]}
  onContextMenu={[Function]}
  onKeyDown={[Function]}
  style={
    Object {
      "WebkitTransitionDuration": "0s",
      "transitionDuration": "0s",
    }
  }
  tabIndex="0"
>
  <div
    className="css-text-901oao"
    dir="auto"
    onClick={[Function]}
    style={
      Object {
        "fontSize": "14px",
      }
    }
  >
    Seleccionar
  </div>
</div>
`;

exports[`Select component render Select component with options 1`] = `
<div
  className="css-view-1dbjc4n r-borderColor-4yh6xw r-borderRadius-1f0042m r-borderWidth-rs99b7 r-cursor-1loqt21 r-display-6koalj r-justifyContent-1777fci r-maxWidth-127gp16 r-paddingBottom-1inuy60 r-paddingLeft-1yt7n81 r-paddingRight-ry3cjt r-paddingTop-glunga r-touchAction-1otgn73 r-transitionProperty-1i6wzkk r-userSelect-lrvibr r-width-13qz1uu"
  onClick={[Function]}
  onContextMenu={[Function]}
  onKeyDown={[Function]}
  style={
    Object {
      "WebkitTransitionDuration": "0s",
      "transitionDuration": "0s",
    }
  }
  tabIndex="0"
>
  <div
    className="css-text-901oao"
    dir="auto"
    onClick={[Function]}
    style={
      Object {
        "fontSize": "14px",
      }
    }
  >
    Seleccionar
  </div>
</div>
`;

exports[`Select component render Select component with value 1`] = `
<div
  className="css-view-1dbjc4n r-borderColor-4yh6xw r-borderRadius-1f0042m r-borderWidth-rs99b7 r-cursor-1loqt21 r-display-6koalj r-justifyContent-1777fci r-maxWidth-127gp16 r-paddingBottom-1inuy60 r-paddingLeft-1yt7n81 r-paddingRight-ry3cjt r-paddingTop-glunga r-touchAction-1otgn73 r-transitionProperty-1i6wzkk r-userSelect-lrvibr r-width-13qz1uu"
  onClick={[Function]}
  onContextMenu={[Function]}
  onKeyDown={[Function]}
  style={
    Object {
      "WebkitTransitionDuration": "0s",
      "transitionDuration": "0s",
    }
  }
  tabIndex="0"
>
  <div
    className="css-text-901oao"
    dir="auto"
    onClick={[Function]}
    style={
      Object {
        "fontSize": "14px",
      }
    }
  >
    select
  </div>
</div>
`;
